File: 02/largest_prime_factor.py
import sys
from math import sqrt,ceil

# returns first factor if not prime or 1 otherwise
# by checking all values up to the sqrt of the value
def min_factor(x):
    for i in range(2,int(ceil(sqrt(x)) + 1)):
        if x % i == 0:
            return i
    return 1

# a function to make things less ugly
def is_prime(x):
    if min_factor(x) == 1:
        return True
    return False

input_val = long(sys.argv[1])

# divide input value by min factor until its prime
guess = input_val / min_factor(input_val)
while True:
    if is_prime(guess):
        print guess
        break
    guess = guess / min_factor(guess)

File: 02/largest_prime_factor.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

long long	largest_prime_factor(long long x);

int		main(int ac, char **av)
{
	if (ac == 2)
		printf("%lu\n", largest_prime_factor((long long)atoll(av[1])));
	return (0);
}

long long	min_factor(long long x)
{
	long long	i;
	long long	range_max;

	i = 1;
	range_max = (long long)ceill(sqrtl((long double)x)) + 1;
	while (++i < range_max)
	{
		if (x % i == 0)
			return (i);
	}
	return (1);
}

int			is_prime(long long x)
{
	return (min_factor(x) == 1);
}

long long	largest_prime_factor(long long x)
{
	long long	guess;

	guess = x / min_factor(x);
	while (1)
	{
		if (is_prime(guess))
			return (guess);
		guess = guess / min_factor(guess);
	}
	return (x);
}
